Formula/smimesign.rb

```diff
@@ -0,0 +1,24 @@
+class Smimesign < Formula
+  desc "S/MIME signing utility for use with Git"
+  homepage "https://github.com/github/smimesign"
+  url "https://github.com/github/smimesign/archive/0.0.3.tar.gz"
+  sha256 "69dfa16737c715679e067d0d929dd964790060b33981695d71bd6b047ae64e97"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/github/smimesign").install buildpath.children
+
+    cd "src/github.com/github/smimesign" do
+      system "go", "build", "-o", bin/"smimesign", "-ldflags", "-X main.versionString=#{version}"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/smimesign --version")
+    system "#{bin}/smimesign", "--list-keys"
+    assert_match "could not find identity matching specified user-id: bad@identity", shell_output("#{bin}/smimesign -su bad@identity 2>&1", 1)
```

Comment:
I wasn't sure if there was a better helper for testing for a negative (non 0 exit) result.